stackdio/core/constants.py

```diff
@@ -64,6 +64,32 @@ class ComponentStatus(object):
     CANCELLED = 'cancelled'
     UNKNOWN = 'unknown'
 
+    @classmethod
+    def aggregate(cls, status_list):
+        # Make sure everything in the list is a valid status
+        assert len([s for s in status_list if s not in vars(cls).values()]) == 0
+
+        if len(status_list) == 0:
+            # We can get an empty list sometimes when we're deleting a stack, so we'll
+            # just aggregate that to unknown
+            return cls.UNKNOWN
+
+        elif cls.FAILED in status_list:
```

Comment:
I'm not sure I like this [`elif`] pattern in the long term (under the assumption we'll wind up adding more statuses later). 

It'd be more future-proof if we assigned each status level a "priority", sorted `status_list` by priority, then took the highest status.